ingest/ingest-datasets-old.py

```diff
@@ -0,0 +1,392 @@
+    for distribution in distributionList:
+        accessURL = str(list(distribution.objects(DCO.accessURL))[0])
+        name = distribution.label().toPython() if distribution else None
+        obj = {"uri": str(distribution.identifier), "accessURL": accessURL, "name": name}
+
+        fileList = list(distribution.objects(DCO.hasFile))
+        # fileList = [f for f in distribution.objects(DCO.hasFile) if has_type(f, DCO.File)]
+        fileList = fileList if fileList else None
+        files = []
+        for file in fileList:
+            downloadURL = list(file.objects(DCO.downloadURL))
+            downloadURL = str(downloadURL[0]) if downloadURL else None
+            fileObj = {"uri": str(file.identifier),
+                       "name": file.label().toPython()}
+            fileObj.update({"downloadURL": downloadURL})
+            files.append(fileObj)
+
+        if files:
+            obj.update({"files": files})
+
+        distributions.append(obj)
+
+    # try:
+    #     authors = sorted(authors, key=lambda a: a["rank"]) if len(authors) > 1 else authors
+    # except KeyError:
+    #     print("missing rank for one or more authors of:", dataset)
+
+    doc.update({"distributions": distributions})
+
+    return doc
+
+
+def has_type(resource, type):
+    for rtype in resource.objects(RDF.type):
+        if str(rtype.identifier) == str(type):
+            return True
+    return False
+
+
+def publish(bulk, endpoint, rebuild, mapping):
+    # if configured to rebuild_index
+    # Delete and then re-create to publication index (via PUT request)
+
+    index_url = endpoint + "/" + _index
+
+    if rebuild:
+        requests.delete(index_url)
+        r = requests.put(index_url)
+        if r.status_code != requests.codes.ok:
+            print(r.url, r.status_code)
+            r.raise_for_status()
+
+    # push current publication document mapping
+
+    mapping_url = index_url + "/" + _type + "/_mapping"
+    with open(mapping) as mapping_file:
+        r = requests.put(mapping_url, data=mapping_file)
+        if r.status_code != requests.codes.ok:
+
+            # new mapping may be incompatible with previous
+            # delete current mapping and re-push
+
+            requests.delete(mapping_url)
+            r = requests.put(mapping_url, data=mapping_file)
+            if r.status_code != requests.codes.ok:
+                print(r.url, r.status_code)
+                r.raise_for_status()
+
+    # bulk import new publication documents
+    bulk_import_url = endpoint + "/_bulk"
+    r = requests.post(bulk_import_url, data=bulk)
+    if r.status_code != requests.codes.ok:
+        print(r.url, r.status_code)
+        r.raise_for_status()
+
+
+def generate(threads, sparql):
+    pool = multiprocessing.Pool(threads)
+    params = [(dataset, sparql) for dataset in get_datasets(endpoint=sparql)]
+    return list(itertools.chain.from_iterable(pool.starmap(process_dataset, params)))
+
+
+if __name__ == "__main__":
```

Comment:
THis is replicated code so we need to figure out a way to put this in a single .py script and call out to a class that implements the specific ingest.